File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> diff = new HashMap<>();
        // difference, 
        for (int i = 0; i < nums.length; i++) {
            diff.put(nums[i], i);
        }

        for (int i = 0; i < nums.length; i++) {
            int difference = target - nums[i];
            if (diff.containsKey(difference) && diff.get(difference) != i) {
                return new int[] {i, diff.get(difference)};
            }
        }
        return new int[0];
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        # two pointers
        n = len(numbers)
        l = 0
        r = n - 1

        while l < r:
            if numbers[r] + numbers[l] == target:
                res = [l+1, r+1]
                return res
            elif numbers[r] + numbers[l] < target:
                l += 1
            else:
                r -= 1